Sources/Confidence/EventStorage.swift

```diff
@@ -0,0 +1,104 @@
+import Foundation
+
+internal protocol EventStorage {
+    func rollover() throws
+    func writeEvent(event: Event) throws
+    func batchReadyFiles() throws -> [URL]
+    func eventsFrom(fileURL: URL) throws -> [Event]
+}
+
+internal class EventStorageImpl: EventStorage {
+    let DIRECTORY = "events"
+    let EVENT_WRITE_DELIMITER = ",\n"
+    let READY_TO_SEND_EXTENSION = ".ready"
+    let encoder = JSONEncoder()
+    let decoder = JSONDecoder()
+    var folderURL: URL = URL(string: "")!
+    var fileURL: URL = URL(string: "")!
+    var currentBatch: [Event] = []
+
+    init() throws {
+        folderURL = try getFolderURL()
+        try createEventsDirectory(url: folderURL)
+        fileURL = folderURL.appendingPathComponent("events-\(Date().currentTime)")
+    }
+
+    func rollover() throws {
+        let newPath = URL(string: "\(fileURL)" + "\(Date().currentTime)")!
+        try FileManager.default.moveItem(at: fileURL, to: newPath)
+        try FileManager.default.removeItem(at: fileURL)
```

Comment:
I am not familiar with `moveItem`, but it sounds like it moves the file to a new location, do we really need to remove it after then?

Reply:
It looks like it, thanks for pointing it out. It doesn't have to be removed.